File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/builder/Builder.java
package jv.com.tencent.cain.design_pattern.builder;

/**
 * @author : jiangyu
 * @date : 2020/2/10
 * @desc : xxx
 */
public abstract class Builder {
    public abstract void buildPartA();

    public abstract void buildPartB();

    public abstract Product getResult();
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/builder/Builder.kt
package kt.com.tencent.cain.desigin_pattern.builder

/**
 * @author : jiangyu
 * @date   : 2020/2/10
 * @desc   : xxx
 */
abstract class Builder {
    abstract fun buildPartA()
    abstract fun buildPartB()
    abstract fun getResult(): Product

}